content/docs/command-reference/exp/branch.md

```diff
@@ -0,0 +1,38 @@
+# exp branch
+
+Commit the results from any [experiment](/doc/command-reference/exp) to Git, in
+a new branch (which will become the current <abbr>workspace</abbr>).
+
+## Synopsis
+
+```usage
+usage: dvc exp branch [-h] [-q | -v] experiment branch
+
+positional arguments:
+  experiment     Experiment to be promoted
+  branch         Git branch name to use
+```
+
+## Description
+
+Makes a Git branch off the last commit (`HEAD`) based on the given `experiment`,
+using the `branch` name provided.
+
+This is equivalent to using `dvc exp apply` (applies the results from any
+experiment to the workspace) followed by Git branching and committing:
```

Comment:
Quick Q on `exp branch`: Is this exactly correct @pmrowla? Thanks

Reply:
The wording should should probably be `roughly equivalent`.

For standalone experiments, `exp branch` is exactly the same as doing:
```
$ dvc exp apply <experiment>
$ git add .
$ git commit
```

For checkpoints there is a difference, and it is actually equivalent to:
```
$ git merge <checkpoint_experiment>
```

The difference is that when you do `exp branch` on a checkpoint experiment, all of the previous checkpoint commits will also be included in the new branch. So for experiment run which contained 5 generated checkpoints, your new branch would append all 5 of those checkpoint commits to the previous HEAD.

Whereas, if you did `exp apply <checkpoint_experiment>` and then git committed the result, your new branch would only have 1 new commit after the previous HEAD. That one commit would be the squashed end result of all 5 of the checkpoint commits.

Reply:
OK, noted, thanks.

> exp branch is exactly the same as doing...

But when would the the `git branch` command happen? Before committing, I assume.

Reply:
`git branch` would be before committing yes, the comparisons are just written assuming that you are already in whichever git branch you want the commits to go.